src/analyses/cfg_dominators.h

```diff
@@ -76,7 +74,45 @@ class cfg_dominators_templatet
   /// Note by definition all program points dominate themselves.
   bool dominates(T lhs, const nodet &rhs_node) const
   {
-    return rhs_node.dominators.count(lhs);
+    if (!rhs_node.dominator)
+      return false;
+
+    auto target_index = cfg.get_node_index(lhs);
+    auto current_index = rhs_node.dominator;
+    while (current_index.has_value())
+    {
+      if (*current_index == target_index)
+        return true;
+
+      // Root dominates itself, so to avoid an endless loop, return.
+      if (*current_index == 0)
+        return false;
+
+      current_index = cfg[*current_index].dominator;
+    }
+
+    return false;
+  }
+
+  /// Returns a set of T that are the dominators of start_node. Only use if you
+  /// need the entire set of dominators from node to root.
+  std::set<T> dominated_by(T start_node) const
```

Comment:
As above, rename `dominators` (but see iterator comment -- `dominators` should return an iterable with a `begin()` and `end()` which walk over the tree)